crates/iceberg/src/spec/manifest.rs

```diff
@@ -79,6 +79,22 @@ impl Manifest {
 
         Ok(Manifest { metadata, entries })
     }
+
+    /// Upgrade the format version of this manifest.
+    pub fn upgrade_version(&mut self, format_version: FormatVersion) -> Result<(), Error> {
```

Comment:
update version seems tricky to me. We don't need to take other actions?

Reply:
+1 for this concern. Version upgrade is part of transaction api, it involves full rewrite of all things like snapshot, and should not do it here.

Reply:
> +1 for this concern. Version upgrade is part of transaction api, it involves full rewrite of all things like snapshot, and should not do it here.

Yes, upgrading the version is more. Here it's more like upgrading this manifest file. https://github.com/apache/iceberg-rust/issues/119#issuecomment-1855355460
This interface is confusing.  There is another way:
1. `pub fn parse_avro(bs: &[u8], version) -> Result<Self, Error>`. Then we can parse v1 manifest as v2. After we parse it, we will modify the version as v2. 
2. `pub async fn write(mut self, manifest: Manifest, version)`. And then we can write as v1 manifest as v2. I think this way may be more clear.🤔

Reply:
I checked that comment, but I don't think we need to introduce this api for now. The manifest format upgradation is part of table format upgradation, and we need to think about it in the overall design, rather exposing this single api.

Reply:
LGTM. Let's remove this interface and only fix the init-default. Let's track the upgrade action in another issue as a new feature.

Reply:
> +1 for this concern. Version upgrade is part of transaction api, it involves full rewrite of all things like snapshot, and should not do it here.

This is not correct, and V2 is forward-compatible. If you apply the rules of schema evolution (as with any other Iceberg table), it should work seamlessly. You don't want to rewrite a multi-petabyte table when you do any meta operation.

Reply:
> Let's remove this interface and only fix the init-default.

This will make thing easier, such as setting the sequence-number when the field isn't present (in the case of V1 metadata).